src/instruction.rs

```diff
@@ -57,6 +57,62 @@ impl fmt::Display for ArithmeticOperator {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub enum LogicalOperand {
+    LiteralInteger(i64),
+    MemoryReference(MemoryReference),
+}
+
+impl fmt::Display for LogicalOperand {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match &self {
+            LogicalOperand::LiteralInteger(value) => write!(f, "{}", value),
+            LogicalOperand::MemoryReference(value) => write!(f, "{}", value),
+        }
+    }
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum LogicalOperator {
+    Binary(BinaryLogic),
+    Unary(UnaryLogic),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum BinaryLogic {
+    And,
+    Or,
+    Ior,
+    Xor,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum UnaryLogic {
+    Neg,
+    Not,
+    True,
+    False,
```

Comment:
The [grammar](https://github.com/quil-lang/quil/blob/master/grammars/Quil.g4#L172-L178) marks `True` and `False` as deprecated; should we still include them?

Reply:
What's more, I don't see them being used elsewhere…